src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -3,384 +3,526 @@
+    this.username = username;
+    this.password = password;
+    this.hostname = hostname;
+    this.port = port;
 
-    public BasicOMEROClient(String username, String password, String hostname, int port) {
+    this.sessionId = "";
 
-        this.username = username;
-        this.password = password;
-        this.hostname = hostname;
-        this.port = port;
+  }
 
-        this.sessionId = "";
+  private void connect(String hostname, int port, String username, String password)
+      throws Exception {
 
+    this.gateway = new Gateway(new SimpleLogger());
+    LoginCredentials cred = new LoginCredentials();
+    cred.getServer().setHostname(hostname);
+
+    if (port > 0) {
+      cred.getServer().setPort(port);
     }
 
-    private void connect(String hostname, int port, String username, String password) throws Exception {
+    cred.getUser().setUsername(username);
+    cred.getUser().setPassword(password);
 
-        this.gateway = new Gateway(new SimpleLogger());
-        LoginCredentials cred = new LoginCredentials();
-        cred.getServer().setHostname(hostname);
+    ExperimenterData user = gateway.connect(cred);
+    this.ctx = new SecurityContext(user.getGroupId());
 
-        if (port > 0) {
-            cred.getServer().setPort(port);
-        }
+    this.sessionId = gateway.getSessionId(user);
+  }
 
-        cred.getUser().setUsername(username);
-        cred.getUser().setPassword(password);
+  public void connect() {
+    try {
 
-        ExperimenterData user = gateway.connect(cred);
-        this.ctx = new SecurityContext(user.getGroupId());
+      this.connect(this.hostname, this.port, this.username, this.password);
+    } catch (Exception e) {
+      System.out.println(e);
+
+      System.out.println("++++++++++++++++++++++++++++++++++++++");
+      System.out.println(e.getLocalizedMessage());
+      System.out.println(e.getMessage());
+      System.out.println(e.toString());
 
-        this.sessionId = gateway.getSessionId(user);
     }
 
-    public void connect() {
-        try {
+  }
+
+  /**
+   * Returns any file annotations (information about attachments) of a given image
+   * 
+   * @param imageID the ID of the image
+   * @return A list of FileAnnotationData objects
+   * @throws DSOutOfServiceException
+   * @throws DSAccessException
+   * @throws ExecutionException
+   */
+  public List<FileAnnotationData> fetchFileAnnotationDataForImage(long imageID)
+      throws DSOutOfServiceException, DSAccessException, ExecutionException {
+    List<FileAnnotationData> annotations = new ArrayList<>();
+    for (AnnotationData d : loadAnnotationsForImage(imageID, FileAnnotationData.class)) {
+      annotations.add((FileAnnotationData) d);
+    }
+    return annotations;
+  }
+
+  /**
+   * Returns any map annotation data (key value pairs of metadata) of a given image
+   * 
+   * @param imageID the ID of the image
+   * @return A list of MapAnnotationData objects
```

Comment:
What are those MapAnnotationData and what is the difference to the previous method?

Reply:
After discussion it would be nice to do this with a generic return type, but since I need to add the type.class (line 144) I think it cannot be done easily, at least not without a parameter "AnnotationType.class". In this case I think the two methods are fine as convenience methods for the user.